datafusion/functions-aggregate-common/src/aggregate.rs

```diff
@@ -0,0 +1,182 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Contains the trait `AggregateExpr` which defines the interface all aggregate expressions
+//! (built-in and custom) need to satisfy.
```

Comment:
As this first sentence is rendered as the summary on the docs page, I think it is nice to make a link to the relevant struct/trait if possible 

for example https://docs.rs/datafusion/latest/datafusion/index.html


![Screenshot 2024-08-08 at 8 27 01 AM](https://github.com/user-attachments/assets/5e201a3b-7e8b-4000-86ce-5a43acb5640b)

```suggestion
//! [`AggregateExpr`] which defines the interface all aggregate expressions
//! (built-in and custom) need to satisfy.
```